### DISCUSSION TITLE
Relay?

### DISCUSSION THREAD

**Discussion by DB1BMN** [flandreas/antares#857] [Ideas]
Hi, how about to add a relay to the analog simulation?
Not just a Multiplexer/Transmission gate but a true galvanic isolated realay.

I think the coil could be easily modelled like the lightbulb - with a coil resistance and an nominal (rated) voltage.
The datasheets usually specifiy a "must operate voltage" and a "must release voltage" in % of nominal voltage, e.g. @ 70% V_nom the relay must turn on and at 10% V_nom the relay must release. So we would have here a true hysteresis.

The contacts should be SPDT or DPDT. With this configuration any other configuration could be built as wanted eg. normally open (NO), normally closed (NC) and arbitrary multiple stages by paralleling the elemant relay "cell".

I think at the moment we do not need any delays or inductive character.

What it would be usefull for? - Zuse Z3, no just kidding but there also some nice educational circuits: Self-Holding Circuit, Star-Delta-Switching, Motor reversal, Elevator-Logic...

What do you think?

**Comment by flandreas** [comment]
Thank you for this suggestion! I was also thinking about implementing an analog relay someday.

I think we should do this properly, that is including the inductive character, i.e. the influence on the current while the magnetic field is building up. It would therefore first implement a stand-alone inductivity (coil element), and then based on that the relay element.

Regarding Zuse Z3: I know you're kidding, but I find that one also fascinating, especially after user @fpw has successfully implemented a PDP-8/I (#738) with Antares. Zuse Z3 would of course not be possible entirely analog, but perhaps the basic element in analog and then using scripting in the higher layers? Maybe there's a niche for Antares in simulating legacy computers..?

**Comment by flandreas** [comment]
Started work on #861 "Inductor (coil)" as basis for a relay.

**Comment by DB1BMN** [comment]
Cool Thanks!
Yes, an inductor would be also nice. Basically its the same behavior like the capacitor except voltage and current are swapped.

Back to the relay: Please be aware that most manufacturers do not specify any inductance for the coil. More worse this is strongly non-linear with voltage (as iron is) and even varies slightly during switching as the air gap changes.
Looking forward for some beta version to test.

Merry X-Mas!
